### PR TITLE
feat: metric engine support alter

### LINKED ISSUE #2866: Procedure to `ALTER` logical table should acquire lock to physical table
To avoid race condition, `ALTER` procedure to logical metric table should acquire the corresponding physical table's lock.

### PR DESCRIPTION
**Pull request by fengjiachun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

1. Support ALTER
2. Procedure to ALTER logical table should acquire lock to physical table

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/common/meta/src/ddl/alter_table.rs [hunk 1]
```diff
@@ -335,13 +353,24 @@ impl AlterTableProcedure {
...
     fn lock_key_inner(&self) -> Vec<String> {
+        let mut lock_key = vec![];
+
+        if let Some(physical_table_name) = self.data.physical_table_name() {
+            let physical_table_key = common_catalog::format_full_table_name(
+                &physical_table_name.catalog_name,
+                &physical_table_name.schema_name,
+                &physical_table_name.table_name,
+            );
+            lock_key.push(physical_table_key);
```
Comment (MichaelScofield) [inline src/common/meta/src/ddl/alter_table.rs:364]: Can we use the physical table id as lockkey (convert it to String of course)? I'm aware physical table name has little chance to be renamed, but it's better to use a hard invariant.

File: src/common/meta/src/ddl/alter_table.rs [hunk 2]
```diff
@@ -182,23 +190,33 @@ impl AlterTableProcedure {
...
+                let physical_table_id = x.physical_table_id();
+                let physical_table_route = table_route_manager
+                    .get(physical_table_id)
+                    .await?
+                    .context(TableRouteNotFoundSnafu {
+                        table_id: physical_table_id,
+                    })?;
+                physical_table_route.region_routes()?.clone()
+            }
+        };
```
Comment (WenyXu) [inline src/common/meta/src/ddl/alter_table.rs:212]: It's okay for now. Let's move these methods into `table_meatdata_manager' later.

File: src/common/meta/src/ddl_manager.rs [hunk 3]
```diff
@@ -327,8 +335,42 @@ async fn handle_alter_table_task(
...
+                    table_name: table_ref.to_string(),
+                })?
+                .table_info;
+            Some(TableName {
+                catalog_name: physical_table_info.catalog_name.clone(),
+                schema_name: physical_table_info.schema_name.clone(),
+                table_name: physical_table_info.name.clone(),
+            })
+        }
+    };
```
Comment (WenyXu) [inline src/common/meta/src/ddl_manager.rs:365]: ditto

**Review by WenyXu — APPROVED** [review]

File: src/common/meta/src/ddl/alter_table.rs (hunk 1 above)
Comment (fengjiachun) [reply src/common/meta/src/ddl/alter_table.rs:364]: We use the table name (Failover and Migration) in other procedures; it is necessary to ensure mutual exclusivity with these.

Comment (fengjiachun) [reply src/common/meta/src/ddl/alter_table.rs:364]: Rename Physical table name is not allowed.

File: src/common/meta/src/ddl/alter_table.rs (hunk 2 above)
Comment (fengjiachun) [reply src/common/meta/src/ddl/alter_table.rs:212]: Let  me do it now

File: src/common/meta/src/ddl/alter_table.rs (hunk 1 above)
Comment (WenyXu) [reply src/common/meta/src/ddl/alter_table.rs:364]: > We use the table name (Failover and Migration) in other procedures; it is necessary to ensure mutual exclusivity with these.

BTW, I will refactor these keys and add more tests recently.

**Comment by fengjiachun** [comment]
@MichaelScofield @WenyXu PTAL

**Review by WenyXu — APPROVED** [review]

**Comment by WenyXu** [comment]
Let's run the `make fmt`

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**